core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
I'm in favor of leaving the `checkNotNull` as-is, but just because I feel the string is useful in and of itself, and that this is a convention used fairly widely (even within Google).

I do, however, disagree with the statement:

> the original author gets to decide and we don't change it later without cause, as it is a style decision well within their right.

This is open source code and the original author is irrelevant.  There is no code ownership and we do not need to consult the original author when tweaking code style.  The only requirement is that any submitted code adhere to the project coding conventions. Anything beyond that is fair game.